fl4health/reporting/base_reporter.py

```diff
@@ -0,0 +1,51 @@
+"""Base Class for Reporters.
+
+Super simple for now but keeping it in a seperate file in case we ass more base methods.
+"""
+
+from typing import Any
+
+
+class BaseReporter:
+    def report(
+        self,
+        data: dict,
+        round: int | None = None,
+        epoch: int | None = None,
+        batch: int | None = None,
+    ) -> None:
+        """A method called by clients or servers to send data to the reporter.
+
+        The report method is called reduncantly by the clients every step, every epoch,
+        every round and sometimes outside of a FL round (for high level summary data).
+        It is up to the reporter to determine when and what to report.
+
+        Args:
+            data (dict): The data to maybe report from the server or client.
+            round (int | None, optional): The current FL round. If None, this indicates
+                that the method was called outside of a round (eg. for summary
+                information). Defaults to None.
+            epoch (int | None, optional): The current epoch. If None then this method
+                was not called within the scope of an epoch. Defaults to None.
+            batch (int | None, optional): The current batch. If None then this method
```

Comment:
Am I missing where we would report metrics on the step level? I assume batch != local steps based on description Or is there a reason we aren't tracking at the step level that I am missing

Reply:
No batch == local steps. I agree the documentation is a bit unclear and needs updating here. The reason I used batch was because step means something else for the wandb reporters, however I agree its inconsistent with the terminology used throughout basic client. Happy to just change it to step if you think that is better